### PR TITLE
[WIP] ebpf: remove bcc macros and types

### PR DESCRIPTION
**Pull request by rootfs**
- [x] remove all bcc macro and function
- [x] pass bcc compile
- [x] pass non-bcc compile
- [ ] integrated with collector

### CODE REVIEW

**Comment by rootfs** [comment]
#716

File: hack/compile-bpf.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+DIR=$(dirname $0)
+CFILE=${CFILE:-"${DIR}/../bpfassets/perf_event/perf_event.c"}
+OUTPUT=${OUTPUT:-"${DIR}/../_output/bpf-modules/perf_event.o"}
+VMLINUX_H=${VMLINUX_H:-"https://raw.githubusercontent.com/iovisor/bcc/master/libbpf-tools/x86/vmlinux_518.h"}
+if [ ! -f ${DIR}/../bpfassets/perf_event/perf_event.c ]; then
+    echo "perf_event.c not found"
+    exit 1
+fi
+if [ ! -f ${DIR}/../_output/bpf/vmlinux.h ]; then
+    echo "downloading vmlinux.h"
```
Comment (marceloamaral) [inline hack/compile-bpf.sh:13]: This looks similar to what libebpf does.

btw, can we build this in one kernel/LinuxOS and run in a different kernel/LinuxOS?

**Comment by rootfs** [comment]
turns out the [bcc elf pkg ](https://github.com/iovisor/gobpf/tree/master/elf) hasn't got any refresh for a long time. Even the pre-compiled module can be loaded, maps, tables, and probes are still missing. Stop working on pre-compile for now. Will revisit it when situations change.

**Comment by marceloamaral** [comment]
I suggest giving the cilium library a shot - it'll definitely make our work a lot easier!

**Pull request closed without merging**